darklang/dark
# Package Types

This adds package types, and uses them to allow us to replace Stdlib/Libs/Http.fs with Dark code:
Replace Http functions with new http package
- the http framework relies on this
- remove DHttpResponse and THttpResponse
- remove Test functions that just extracted fields from DHttpResponses

Parser changes:
- any usage of Package functions or types must now be fully qualified: `PACKAGE.{owner}.{module}.{fn}` ("PACKAGE" here is the literal string "PACKAGE")
- Rename availableTypes to Types everywhere
- rename completeParse to resolveNames

Test changes:
- no longer parse package definitions in .test files
- switch http tests to new signature

Types changes:
- route all type finding through `RuntimeTypes.Types.find`

Misc:
- Remove Json.parse Record/Dict match
- Fix some issues with closing down the pubsub emulator (was using all my memory)
- ~move testing to a separate step in CI, allowing faster reruns~ - couldn't make this work
